clippy_utils/src/lib.rs

```diff
@@ -1224,6 +1224,68 @@ pub fn get_parent_as_impl(tcx: TyCtxt<'_>, id: HirId) -> Option<&Impl<'_>> {
     }
 }
 
+/// Removes blocks around an expression, only if the block contains just one expression
+/// and no statements.
+///
+/// Examples:
+///  * `{}`       -> `{}`
+///  * `{ x }`    -> `x`
+///  * `{{ x }}`  -> `x`
+///  * `{ x; }`   -> `{ x; }`
+///  * `{ x; y }` -> `{ x; y }`
```

Comment:
I would like one more example that shows that `unsafe` blocks will not be peeled. The same for the other function.


```rs
///  * `{ unsafe { x } }` -> `unsafe { x }`
```